Edits to the remaining index pages as part of site review (#205)
pinot-contrib/pinot-docs: basics/indexing/json-index.md

---
description:  This page describes configuring the JSON index for Apache Pinot.
---

# JSON Index

The JSON index can be applied to JSON string columns to accelerate value lookups and filtering for the column.

## When to use JSON index

> JSON string can be used to represent array, map, and nested fields without forcing a fixed schema. It is flexible, but the flexibility comes with a cost - filtering on JSON string columns is very expensive.

Reviewer: ```suggestion
Use the JSON string to represent array, map, and nested fields without forcing a fixed schema. While JSON strings are flexible, filtering on JSON strings is expensive, so consider the use case.
```

}
```

> Without an index, in order to look up key and filter records based on the value, we need to scan and reconstruct the JSON object from the JSON string for every record, look up the key and then compare the value.

Reviewer: ```suggestion
Without an index, to look up the key and filter records based on the value, Pinot must scan and reconstruct the JSON object from the JSON string for every record, look up the key and then compare the value.
```

}
```

Note that the JSON index can only be applied to `STRING/JSON` columns whose values are JSON strings.

{% hint style="info" %}

> When you're using a JSON index, it is recommended that you add the indexed column to the `noDictionaryColumns` columns list to reduce unnecessary storage overhead.

Reviewer: ```suggestion
When you're using a JSON index, we recommend that you add the indexed column to the `noDictionaryColumns` columns list to reduce unnecessary storage overhead.
```

Reviewer: or perhaps:
To reduce unnecessary storage overhead when using a JSON index, add the indexed column to the `noDictionaryColumns` columns list.
